Lib/fontTools/merge.py

```diff
@@ -363,6 +385,68 @@ def _glyphsAreSame(glyphSet1, glyphSet2, glyph1, glyph2):
 		g1.width == g2.width and
 		(not hasattr(g1, 'height') or g1.height == g2.height))
 
+@_add_method(ttLib.getTableClass('CFF '))
+def merge(self, m, tables):
+	newcff = tables[0]
+	newfont = newcff.cff[0]
+	private = newfont.Private
+	storedNamesStrings = []
+	glyphOrderStrings = []
+	glyphOrder = set(newfont.getGlyphOrder())
+	for name in newfont.strings.strings:
+		if name not in glyphOrder:
+			storedNamesStrings.append(name)
+		else:
+			glyphOrderStrings.append(name)
+	chrset = list(newfont.charset)
+	newcs = newfont.CharStrings
+	newcsi = newcs.charStringsIndex
+	log.info("Font 0 global subrs: %s.", len(newcff.cff.GlobalSubrs))
+	ls = None
+	if hasattr(newfont, 'Private') and hasattr(newfont.Private, 'Subr'):
+		ls = newfont.Private.Subrs
+	lenls = len(ls) if ls is not None else 0
+	log.info("Font 0 local subrs: %s.", lenls)
+	log.info("FONT 0 CharStrings: %s, %s.",(len(newcs), len(newcsi)))
+	baseIndex = len(newcsi)
+	for i, table in enumerate(tables[1:], start=1):
+		font = table.cff[0] 
+		font.Private = private
+		fontGlyphOrder = set(font.getGlyphOrder())
+		for name in font.strings.strings:
+			if name in fontGlyphOrder:
+				glyphOrderStrings.append(name)
+		cs = font.CharStrings
+		gs = table.cff.GlobalSubrs
+		log.info("Font %s global subrs: %s." % (str(i), str(len(gs))))
+		ls = None
+		if hasattr(font, 'Private') and hasattr(font.Private, 'Subr'):
+			ls = font.Private.Subrs
+		lenls = len(ls) if ls is not None else 0
```

Comment:
@BlackFoundry you're getting the local subroutines from the Private dict just to print the length, but you don't actually do anything with it.. Same for the global subroutines. :thinking:

Reply:
yes you're right, that was for testing and informative purposes actually, they are not used for anythng else.

Reply:
should we actually raise an error if they are not empty? The code expects the cff tables to be desubroutinized